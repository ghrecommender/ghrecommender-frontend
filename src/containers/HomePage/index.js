// @flow strict
import React from 'react'

import { Button, Container, Header, Icon, Menu, Segment, Responsive } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import Footer from '../../components/Footer'
import Background from '../../components/Background'
import StatisticGroups from '../../components/Statistics'
import { trackClick } from '../../utils'

const logoStyle = {
  marginTop: '3em',
  userSelect: 'none',
  userDrag: 'none',
  webkitUserDrag: 'none',
}

const renderHeaders = (mobile: boolean) => {
  const fontSize1 = mobile ? '2em' : '4em'
  const fontSize2 = mobile ? '1.5em' : '1.7em'
  return (
    <div>
      <img alt="logo" src="/assets/android-chrome-192x192.png" style={logoStyle} />
      <Header
        as="h1"
        content="GHRecommender"
        inverted
        style={{ fontSize: fontSize1, fontWeight: 'normal', marginBottom: 0 }}
      />
      <Header
        as="h2"
        content="Welcome to the GHRecommender project – the place where you can find the GitHub projects that may be of interest to you but you haven’t discovered yet."
        inverted
        style={{ fontSize: fontSize2, fontWeight: 'normal', marginBottom: '1.7em' }}
      />
    </div>
  )
}

const HomePage = () => (
  <div>
    <Segment
      inverted
      textAlign="center"
      style={{ padding: '1em 0em', minHeight: '100vh' }}
      vertical
    >
      <Container style={{ zIndex: 999, position: 'inherit' }}>
        <Menu inverted pointing secondary size="large">
          <Menu.Item as={Link} to="/" active>
            Home
          </Menu.Item>
          <Menu.Item
            as="a"
            href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
            onClick={trackClick('Menu: How it works')}
          >
            How it Works
          </Menu.Item>
        </Menu>
      </Container>

      <Container text style={{ zIndex: 999, position: 'inherit' }}>
        <Responsive minWidth={700}>{renderHeaders(false)}</Responsive>
        <Responsive maxWidth={700}>{renderHeaders(true)}</Responsive>
        <Button
          primary
          size="huge"
          as={Link}
          to="/app/recommendations/"
          onClick={trackClick('Get Recommendations')}
        >
          Get Recommendations
          <Icon name="right arrow" />
        </Button>
        <Header as="h3" inverted>
          Want to know more? Read check out the{' '}
          <a
            href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
            color="white"
          >
            “How it Works”
          </a>{' '}
          page.
        </Header>
      </Container>

      <Container text style={{ paddingTop: '3em', paddingBottom: '3em' }}>
        <StatisticGroups />
      </Container>

      <Background />
    </Segment>
    <Footer />
  </div>
)

export default HomePage
