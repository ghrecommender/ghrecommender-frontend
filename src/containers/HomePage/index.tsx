import { CSSProperties } from 'react'

import { Button, Container, Header, Icon, Menu, Segment } from 'semantic-ui-react'
import Link from 'next/link'

import Footer from '@/components/Footer'
import StatisticGroups from '@/components/Statistics'
import { trackClick } from '@/utils'

import dynamic from 'next/dynamic'

const Background = dynamic(() => import('@/components/Background'), {
  ssr: false,
})

const logoStyle = {
  marginTop: '3em',
  userSelect: 'none',
} satisfies CSSProperties

const Headers = () => {
  return (
    <div className='heading'>
      <img alt="logo" src="/assets/android-chrome-192x192.png" style={logoStyle} />
      <Header
        as="h1"
        className='h1'
        content="GHRecommender"
        inverted
        style={{ fontWeight: 'normal', marginBottom: 0 }}
      />
      <Header
        as="h2"
        className='h2'
        content="Welcome to the GHRecommender project ‚Äì the place where you can find the GitHub projects that may be of interest to you but you haven‚Äôt discovered yet."
        inverted
        style={{ fontWeight: 'normal', marginBottom: '1.7em' }}
      />
    </div>
  )
}

const HomePage = () => (
  <div>
    <Segment
      inverted
      textAlign="center"
      style={{ padding: '1em 0em', minHeight: '100vh' }}
      vertical
    >
      <Container style={{ zIndex: 999, position: 'inherit' }}>
        <Menu inverted pointing secondary size="large">
          <Menu.Item as={Link} href="/" active>
            Home
          </Menu.Item>
          <Menu.Item
            as="a"
            href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
            onClick={trackClick('Menu: How it works')}
            target="_blank"
          >
            How it Works
          </Menu.Item>
        </Menu>
      </Container>

      <Container text style={{ zIndex: 999, position: 'inherit' }}>
        <Headers />
        <Button
          primary
          size="huge"
          as={'button'}
          onClick={trackClick('Get Recommendations')}
          disabled
        >
          Get Recommendations
          <Icon name="right arrow" />
        </Button>
        <p><i>Service is no longer available üò¢</i></p>
        <Header as="h3" inverted>
          Want to know more? Read check out the{' '}
          <a
            href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
            color="white"
          >
            ‚ÄúHow it Works‚Äù
          </a>{' '}
          page.
        </Header>
      </Container>

      <Container text style={{ paddingTop: '3em', paddingBottom: '3em' }}>
        <StatisticGroups />
      </Container>

      <Background />
    </Segment>
    <Footer />
  </div>
)

export default HomePage
