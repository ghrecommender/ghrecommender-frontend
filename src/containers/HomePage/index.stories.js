import React from 'react'
import { MemoryRouter } from 'react-router-dom'

import HomePage from './index'

export default {
  title: 'Pages|Home',
  decorators: [story => <MemoryRouter>{story()}</MemoryRouter>],
}

export const homePage = () => <HomePage />
