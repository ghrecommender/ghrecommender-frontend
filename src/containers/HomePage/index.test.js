import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import HomePage from './index'

test('<HomePage />', () => {
  const component = renderer.create(
    <MemoryRouter>
      <HomePage />
    </MemoryRouter>,
  )
  expect(component.toJSON()).toMatchSnapshot()
})
