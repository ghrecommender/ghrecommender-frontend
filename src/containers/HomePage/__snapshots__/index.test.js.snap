// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> 1`] = `
<div>
  <div
    className="ui inverted vertical center aligned segment"
    style={
      Object {
        "minHeight": "100vh",
        "padding": "1em 0em",
      }
    }
  >
    <div
      className="ui container"
      style={
        Object {
          "position": "inherit",
          "zIndex": 999,
        }
      }
    >
      <div
        className="ui large inverted pointing secondary menu"
      >
        <a
          className="active item"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="item"
          href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
          onClick={[Function]}
        >
          How it Works
        </a>
      </div>
    </div>
    <div
      className="ui text container"
      style={
        Object {
          "position": "inherit",
          "zIndex": 999,
        }
      }
    >
      <div>
        <div>
          <img
            alt="logo"
            src="/assets/android-chrome-192x192.png"
            style={
              Object {
                "marginTop": "3em",
                "userDrag": "none",
                "userSelect": "none",
                "webkitUserDrag": "none",
              }
            }
          />
          <h1
            className="ui inverted header"
            style={
              Object {
                "fontSize": "4em",
                "fontWeight": "normal",
                "marginBottom": 0,
              }
            }
          >
            GHRecommender
          </h1>
          <h2
            className="ui inverted header"
            style={
              Object {
                "fontSize": "1.7em",
                "fontWeight": "normal",
                "marginBottom": "1.7em",
              }
            }
          >
            Welcome to the GHRecommender project – the place where you can find the GitHub projects that may be of interest to you but you haven’t discovered yet.
          </h2>
        </div>
      </div>
      <a
        className="ui huge primary button"
        href="/app/recommendations/"
        onClick={[Function]}
        role="button"
      >
        Get Recommendations
        <i
          aria-hidden="true"
          className="right arrow icon"
          onClick={[Function]}
        />
      </a>
      <h3
        className="ui inverted header"
      >
        Want to know more? Read check out the
         
        <a
          color="white"
          href="https://medium.com/@andrey_lisin/building-recommender-system-for-github-a8108f0cb2bd"
        >
          “How it Works”
        </a>
         
        page.
      </h3>
    </div>
    <div
      className="ui text container"
      style={
        Object {
          "paddingBottom": "3em",
          "paddingTop": "3em",
        }
      }
    >
      <div
        className="ui small inverted statistics"
      >
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            19,925,204
          </div>
          <div
            className="label"
          >
            Users
          </div>
        </div>
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            64,245,061
          </div>
          <div
            className="label"
          >
            Repositories
          </div>
        </div>
        <div
          className="ui statistic"
        >
          <div
            className="value"
          >
            81,302,007
          </div>
          <div
            className="label"
          >
            Stars
          </div>
        </div>
      </div>
    </div>
    <div>
      <canvas
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui inverted vertical segment"
    style={
      Object {
        "padding": "5em 0em",
      }
    }
  >
    <div
      className="ui container"
    >
      <div
        className="ui inverted divided grid"
      >
        <div
          className="row"
        >
          <div
            className="six wide column"
          >
            <h4
              className="ui inverted header"
            >
              About
            </h4>
            <div
              className="ui inverted link list"
              role="list"
            >
              <a
                className="item"
                href="mailto:hire@ghrecommender.io"
                onClick={[Function]}
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  className="envelope outline icon"
                  onClick={[Function]}
                />
                 Hire us
              </a>
              <a
                className="item"
                href="https://github.com/ghrecommender"
                onClick={[Function]}
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  className="github icon"
                  onClick={[Function]}
                />
                 GitHub
              </a>
              <a
                className="item"
                href="https://twitter.com/ghrecommender"
                onClick={[Function]}
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  className="twitter icon"
                  onClick={[Function]}
                />
                 Twitter
              </a>
            </div>
          </div>
          <div
            className="six wide column"
          >
            <p>
              Copyright © 2017 
              <a
                href="https://github.com/avli"
              >
                Andrey Lisin
              </a>
               and
               
              <a
                href="http://github.com/yurtaev"
              >
                Egor Yurtaev
              </a>
            </p>
            <p>
              The code is licensed under MIT.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
